### DISCUSSION TITLE
[Feature] Search-By-Image for Faces and Smart Search

### DISCUSSION THREAD

**Discussion by Broyojo** [immich-app/immich#13346] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Since Immich uses CLIP and facial recognition models for SmartSearch and facial recognition respectively, it would be great to utilize the other search aspect that these models allow, which would be "search-by-image", where we can supply an image and search for similar images with SmartSearch, or we can supply a face and find the most similar faces in the database. This would help when you can't accurately describe an image, or especially in the case of a face when you just have a reference image of the person's face. 

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#4509
